ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -1778,6 +1778,20 @@ static IHqlExpression * simplifySortlistComplexity(IHqlExpression * sortlist)
                     appendComponent(cpts, invert, &concats.item(idxc));
             }
         }
+        else if (cur->getOperator() == no_trim)
+        {
+            //TRIM(fixed-length-string) can just sort by the string instead.  (Don't match LEFT/RIGHT versions.)
```

Comment:
Seems to me the root of the dilemma is that 'ab' and 'ab  '  when stored in variable length strings compare equal yet their lengths do not. Which is weird.

I suspect the best bet is to ignore it, SO LONG AS we only implicitly mark sorts UNSTABLE if the user did not explicitly request stable.